examples/manual_reply/test/test.ts

```diff
@@ -10,77 +12,295 @@ const manual_reply_canister = createActor('rrkah-fqaaa-aaaaa-aaaaq-cai', {
 const tests: Test[] = [
     ...cleanDeploy('manual_reply'),
     {
-        name: 'manual_query when calling ic.reject',
+        name: 'manual_update when calling ic.reject',
         test: async () => {
             const rejectionMessage = 'reject';
             try {
-                await manual_reply_canister.manual_query(rejectionMessage);
+                await manual_reply_canister.manual_update(rejectionMessage);
 
                 return {
                     ok: false
                 };
             } catch (error) {
                 return {
                     ok:
-                        (error as any).props.Code === 'CanisterReject' &&
-                        (error as any).props.Message === rejectionMessage
+                        (error as Error).message.includes('Reject code: 4') &&
+                        (error as Error).message.includes(
+                            `Reject text: ${rejectionMessage}`
+                        )
                 };
             }
         }
     },
     {
-        name: 'manual_query when calling ic.reply',
+        name: 'manual_update when calling ic.reply',
         test: async () => {
-            // TODO: Once ic.reply is implemented this shouldn't throw
-            try {
-                await manual_reply_canister.manual_query('accept');
+            const result = await manual_reply_canister.manual_update('accept');
 
-                return {
-                    ok: false
-                };
-            } catch (error) {
-                return {
-                    ok: (error as any).props.Code === 'CanisterError'
-                };
-            }
+            return {
+                ok: result === 'accept'
+            };
         }
     },
     {
-        name: 'manual_update when calling ic.reject',
+        name: 'update reply with blob',
+        test: async () => {
+            const result = await manual_reply_canister.update_blob();
+            const expectedResult = [83, 117, 114, 112, 114, 105, 115, 101, 33];
+
+            return {
+                ok:
+                    Array.isArray(result) &&
+                    result.every((byte, i) => byte === expectedResult[i])
+            };
+        }
+    },
+    {
+        name: 'update reply with float32',
+        test: async () => {
+            // const result = await manual_reply_canister.update_float32();
+            // Note: The JS agent doesn't handle floats correctly.
+            // See https://github.com/dfinity/agent-js/issues/589
+            const result = execSync(
+                'dfx canister call manual_reply query_float32'
```

Comment:
Shouldn't this be a call to `update_float32`?

Reply:
Good catch. This was definitely just copy/pasta so good eye. I've pushed a change.